www/guides/react.md

```diff
@@ -0,0 +1,373 @@
+
+In this guide, you'll go from a [minimal Snowpack project](https://github.com/pikapkg/snowpack/tree/master/create-snowpack-app/app-template-minimal) to a Snowpack project with React and several useful developer optimizations. In the process you'll learn:
+
+- How to set up a new Snowpack project
+- How to get started with React in a Snowpack project
+- Importing and using CSS and other assets inside your project
+- Enabling [Fast Refresh](https://reactnative.dev/docs/fast-refresh) mode for React
+- Extending Snowpack with plugins
+
+> If you want to jump to the end to see a full featured React setup, the Create Snowpack App React template comes with everything you'll learn in this tutorial plus other useful tools.
+
+Prerequisites: Snowpack is a command line tool installed from npm. This guide assumes a basic understanding of Node.js, npm, and how to run commands in the terminal. Knowledge of React is not required, Snowpack is a great way to learn React.
+
+## Getting started with Snowpack and React
+
+In this step by step you'll learn how to use the [`create-snowpack-app`](https://www.snowpack.dev/#create-snowpack-app-(csa) tool to spin up a working Snowpack project and then how to add React to it.
+
+`create-snowpack-app` is a command line tool that creates a working Snowpack App based on a template. The template you're starting with here is `@snowpack/project-template-minimal is` a bare-bones setup where Snowpack serves an index.html, index.css, and index.js.
+
+```bash
+npx create-snowpack-project react-snowpack --template @snowpack/project-template-minimal
+cd react-snowpack
+npm run start
+```
+
+Image: screenshot of project-template-minimal, which shows "Hello world" in text on a white background
+
+This is about as basic as it gets. When you run `npm start` Snowpack serves an `index.html`, `index.css`, and `index.js`.
+
+> The `README.md` contains useful information about what each file does.
+
+The first thing that you'll want to do is install React in your project. To do that, run the following command in your project directory:
+
+To install these you can run:
+
+```bash
+npm install react react-dom --save
+```
+
+> In this tutorial commands use `--save` for frontend libraries and `--save-dev` for your developer tools to keep them organized separately. This is totally up to you, though.
+
+## Create your first React component
+
+React uses a syntax called [JSX](https://reactjs.org/docs/introducing-jsx.html) for components. Snowpack has built in support for JSX files with the extension `.jsx`. To create your first React component in Snowpack rename `index.js` to `index.jsx`.
+
+> You may have used JSX in `.js` files before, but Snowpack expects JSX to be .jsx files
```

Comment:
I think the comment I made above means you don't need to hit on this point twice.